nns/nns_contract.go

```diff
@@ -468,20 +472,29 @@ func putNameStateWithKey(ctx storage.Context, tokenKey []byte, ns NameState) {
 	storage.Put(ctx, nameKey, nsBytes)
 }
 
-// getRecord returns domain record.
-func getRecord(ctx storage.Context, tokenId []byte, name string, typ RecordType) string {
-	recordKey := getRecordKey(tokenId, name, typ)
-	recBytes := storage.Get(ctx, recordKey)
-	if recBytes == nil {
-		return recBytes.(string) // A hack to actually return NULL.
+// getRecords returns domain record.
+func getRecords(ctx storage.Context, tokenId []byte, name string, typ RecordType) []string {
+	recordsKey := getRecordKey(tokenId, name, typ)
+
+	var result []string
+	records := storage.Find(ctx, recordsKey, storage.DeserializeValues)
+	for iterator.Next(records) {
+		r := iterator.Value(records).(struct {
+			key string
+			rs  RecordState
+		})
+		value := r.rs.Data
+		rTyp := r.key[len(r.key)-5]
+		if rTyp == byte(typ) {
+			result = append(result, value)
+		}
 	}
-	record := std.Deserialize(recBytes.([]byte)).(RecordState)
-	return record.Data
+	return result
 }
 
 // putRecord stores domain record.
 func putRecord(ctx storage.Context, tokenId []byte, name string, typ RecordType, record string) {
-	recordKey := getRecordKey(tokenId, name, typ)
+	recordKey := addUniqueBytesToRecordKey(getRecordKey(tokenId, name, typ))
```

Comment:
How do I replace an existing record with this API? I expect `SetRecord` (e.g. of both A and CNAME type) to replace old record, not to add new IP along with the old (possibly invalid one). I think `AddRecord` can use this logic and `SetRecord` (`SetRecords` to replace a set of records).